### DISCUSSION TITLE
Row Level Security blocks all requests

### DISCUSSION THREAD

**Discussion by dostuffthatmatters** [supabase/supabase#2714] [Questions]
I am trying to allow certain users to insert stuff into a table. Since I do not want to share the project secret this should be based on email/password credentials.

The sign-in works fine and the `supabase.auth.user()` object immediately before the request is correct. The code I am using:

https://github.com/dostuffthatmatters/esm-air-quality-rt-plot-data/blob/4b923384854a85e2d6f317183f1f882455933199/run.py#L43-L61

I have tried any combination of RLS rules but comparisons with `role() = 'authenticated'` or `email() = '...'` always result in a blocked request. Has anyone here used Python and RLS before?

**Comment by inian** [comment]
Can you share the exact policies you are using @dostuffthatmatters? If it is confidential, reach out to us via support@supabase.io
